packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -492,6 +494,9 @@ export class DataMenu extends React.PureComponent {
                 : null
             }
             useFullRedirectsForLinks={this.props.useFullRedirectsForLinks}
+            onClick={() =>
+              this.props.onMenuItemClick && this.props.onMenuItemClick(menu)
```

Comment:
You already check this `else this.props.onClick && this.props.onClick();`

```suggestion
              this.props.onMenuItemClick(menu)
```